kafka-admin/src/main/resources/graphql-schema/kafka-admin.graphql

```diff
@@ -1,11 +1,67 @@
-extend type Query {
-    topic(topicName: String): Topic
-    topicList(filter: String): [Topic]
+type TopicDescription {
+    #id: ID
+    name: String
+
+    """@transient"""
+    partitions: [Partitions]
+    config: TopicConfig
+}
+
+type Partitions {
```

Comment:
Trivial/Opiniated
There are two class of types in GraphQL;

- Embeeded types
- Root (Query and Mutation return types)

I tend to organize schema to add extra comment that this type is not used in queries/mutations.

""" embedded """